FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java

```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
     public void findMatch() {
         SharedPreferences preferences = getSharedPreferences("PREFS", 0);
         String[] fileNameList = preferences.getString("fileNameList", "").split(" ");
+        ArrayList<Pair<String, Integer>> matches = new ArrayList<>();
```

Comment:
Because the adapter for the listview uses an (array)list. The adapter also needs to be able to get the items in a particular way to be able to display them.